### PR TITLE
[ACR] SystemData and new readonly KeyVaultProperties

### PR DESCRIPTION
**Pull request by hkuadithya**
1. Adding `SystemData` definition to Resource and ProxyResource definitions to the preview api versions.
2. Adding a couple of readonly properties to the KeyVaultProperties definition in 2019-12-01-preview version.

---

<i>MSFT employees can try out our new experience at <b>[OpenAPI Hub](https://aka.ms/openapiportal) </b> - one location for using our validation tools and finding your workflow. 
</i>

### Changelog
Please ensure to add changelog with this PR by answering the following questions.
  1. What's the purpose of the update?    
      - [ ] new service onboarding 
      - [ ] new API version 
      - [ ] update existing version for new feature 
      - [ ] update existing version to fix swagger quality issue in s360
      - [ ] Other, please clarify 
  2. When you are targeting to deploy new service/feature to public regions? Please provide date, or month to public if date is not available yet.
  3. When you expect to publish swagger? Please provide date, or month to public if date is not available yet.
  4. If it's an update to existing version,  please select SDKs of specific language and CLIs that require refresh after swagger is published.
      - [ ] SDK of .NET (need service team to ensure code readiness)
      - [ ] SDK of Python
      - [ ] SDK of Java
      - [ ] SDK of Js
      - [ ] SDK of Go
      - [ ] PowerShell
      - [ ] CLI
      - [ ] Terraform
      - [ ] No, no need to refresh for updates in this PR

### Contribution checklist:
- [x] I commit to follow the [Breaking Change Policy](http://aka.ms/bcforapi) of “no breaking changes
- [ ] I have reviewed the [documentation](https://aka.ms/ameonboard) for the workflow.
- [ ] [Validation tools](https://aka.ms/swaggertools) were run on swagger spec(s) and errors have all been fixed in this PR. [How to fix?](https://aka.ms/ci-fix)

If any further question about AME onboarding or validation tools, please view the [FAQ](https://aka.ms/faqinprreview).

### ARM API Review Checklist
- [ ] Ensure to check this box if one of the following scenarios meet updates in the PR, so that label “WaitForARMFeedback” will be added automatically to involve ARM API Review. Failure to comply may result in delays for manifest application. Note this does not apply to data plane APIs, all “removals” and “adding a new property” no more require ARM API review.
  - Adding new API(s)
  - Adding a new API version
  - Adding a new service

- [ ] Please ensure you've reviewed following [guidelines](https://aka.ms/rpguidelines) including [ARM resource provider contract](https://github.com/Azure/azure-resource-manager-rpc) and [REST guidelines](https://github.com/microsoft/api-guidelines/blob/vNext/azure/Guidelines.md). Estimated time (4 hours). This is required before you can request review from ARM API Review board.

- [ ] If you are blocked on ARM review and want to get the PR merged with urgency, please get the ARM oncall for reviews (*RP Manifest Approvers* team under <ins>Azure Resource Manager service</ins>) from IcM and reach out to them. 

### Breaking Change Review Checklist 
If there are following updates in the PR, ensure to request an approval from API Review Board as defined in the [Breaking Change Policy](http://aka.ms/bcforapi). 

- [ ] Removing API(s) in stable version
- [ ] Removing properties in stable version
- [ ] Removing API version(s) in stable version
- [ ] Updating API in stable version with Breaking Change Validation errors
- [ ] Updating API(s) in preview over 1 year

To initiate an evaluation of the breaking change, create a new intake using the [template for breaking changes](https://msazure.visualstudio.com/One/_workitems/create/Scenario?templateId=d1c215ec-5d6a-4cb3-a193-95ddd0a1ba96&ownerId=85bb92fb-21fe-41ca-9309-47f2c5f57515). Addition details on the process and office hours are on the [Breaking change Wiki](https://dev.azure.com/msazure/AzureWiki/_wiki/wikis/AzureWiki.wiki/37684/Breaking-Changes).

Please follow the link to find more details on [PR review process](https://aka.ms/SwaggerPRReview).

### CODE REVIEW

**Comment by hkuadithya** [comment]
@jaysterp @adewaleo @yugangw-msft could you please review this PR?

**Comment by savaradh** [comment]
LGTM

**Comment by adewaleo** [comment]
looks good to me as well. I think these properties need to be added to the 2020-11-01-preview api version being added[here](2020-11-01-preview). Also assuming the api versions being updated are the only ones that support system data?

**Comment by fengzhou-msft** [comment]
@hkuadithya can you fix the ModelValidation error following the [fix guide](https://github.com/Azure/azure-rest-api-specs/blob/master/documentation/ci-fix.md#model-validation)?

**Comment by hkuadithya** [comment]
@fengzhou-msft After adding SystemData, all the local validation errors using OAV asked us to remove location and tags root level properties.
Therefore, as a POC, we remove the [location and tags from AgentPool in this Commit](https://github.com/Azure/azure-rest-api-specs/pull/11601/commits/542e3cc5ac45e942a3e91449a55db8a36aad1dc5). After this, the validation errors reduced from 106 to 98. `oav validate-example <swagger-spec>`

Is there a mistake in the validation? Why are we being asked to remove location and tags root level properties upon introduction of SystemData? Can you please advise us on how to proceed with this? We've only added few readOnly properties. In addition, none of the proxy resources seem to be raising errors, only tracked resources.

cc @huanwu @yugangw-msft

File: specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-05-01-preview/containerregistry_scopemap.json [hunk 1]
```diff
@@ -886,26 +913,58 @@
         }
       }
     },
-    "ProxyResource": {
-      "description": "The resource model definition for a ARM proxy resource. It will have everything other than required location and tags.",
+    "SystemData": {
```
Comment (hkuadithya) [inline specification/containerregistry/resource-manager/Microsoft.ContainerRegistry/preview/2019-05-01-preview/containerregistry_scopemap.json:916]: Reason for adding local SystemData model:
OAV tool has a bug which causes model validation failure upon referencing SystemData from ARM common-types.
Therefore, ARM team has suggested using local models to circumvent this issue.
Bug: https://devdiv.visualstudio.com/DevDiv/_workitems/edit/1085423, 
https://github.com/Azure/oav/issues/280

**Review by fengzhou-msft — APPROVED** [review]

**Pull request merged**